batch_worker.py

```diff
@@ -0,0 +1,143 @@
+from prometheus_api_client import *
+import csv
+from logger import *
+from queries import *
+from pandas import DataFrame,concat,date_range
+GB_FACTOR = 1e9
+class batch_worker(object):
+    def __init__(self,prom_conn,name,endpoint):
+        self.p_client=prom_conn
+        self.name=name
+        self.endpoint=endpoint
+        self.endpointLogger = Logger("endpoint"+name,formatter,logPath).get_logger(name+".log",logging.INFO)
+
+    def get_metric_data(self,query, epoch: int = None):
+        parameters = {"query": query}
+        if epoch:
+            # For range date, values before epoch is considered for aggregation
+            parameters.update({"params": {"time": epoch}})
+        metric_data = self.p_client.custom_query(**parameters)
+        self.endpointLogger.info("get_metric_data %s" , str(parameters))
+        metric_data = self.p_client.custom_query(**parameters)
+    
+        return DataFrame(
+            [
+                {**item["metric"], "value": float(item["value"][1])}
+                for item in metric_data
+            ]
+        )
+
+    def get_grouped_aggregation(self,df_data_list, agg_func: str = "max"):
+        return concat(df_data_list).groupby(
+            [ID_LABEL, "namespace"]
+        ).agg(agg_func).reset_index()
+
+    def get_merged_data(self,df_data1, df_data2):
+        return df_data1.merge(df_data2, on=[ID_LABEL, "namespace"], how="left")
+    
+    def get_request_data_batch(self,end_date, cluster_ids_filter):
+        df_request_data = self.process_metrics_batch(
+            RESOURCE_REQUEST, cluster_ids_filter, end_date.strftime("%s")
+        )
+
+        df_request_data = df_request_data.pivot(
+            index=[ID_LABEL, "namespace"],
+            columns="resource",
+        ).swaplevel(1, 0, axis=1)
+
+        df_request_data.columns = ["_".join(col) for col in df_request_data.columns]
+        df_request_data.memory_value = df_request_data.memory_value / GB_FACTOR
+
+        return df_request_data
+    
+    def get_usage_data_batch(self,start_date, end_date, cluster_ids_filter):
+        cpu_usage_max = []
+        memory_usage_max = []
+
+        for dt in date_range(start_date, end_date):
+            epoch = dt.strftime("%s")
+
+            cpu_usage_max.append(
+                self.process_metrics_batch(CPU_USAGE_MAX, cluster_ids_filter, epoch)
+            )
+            memory_usage_max.append(
+                self.process_metrics_batch(MEMORY_USAGE_MAX, cluster_ids_filter, epoch)
+            )
+
+        df_cpu_usage_max = self.get_grouped_aggregation(cpu_usage_max, "max")
+        df_memory_usage_max = self.get_grouped_aggregation(memory_usage_max, "max")
```

Comment:
Are these doing a max of max? The query over which it max-es already includes a max_over_time, right?

Reply:
@bjoydeep This is to get the max from each of the days MAX.